packages/dcos-sdk/src/NavigationService.ts

```diff
@@ -0,0 +1,80 @@
+import { injectable, inject, named } from "inversify";
+import { EventEmitter } from "events";
+import { IExtensionProvider, ExtensionProvider } from "plugin-baby";
+
+const NAVIGATION_CHANGE = Symbol("NAVIGATION_CHANGE");
+
+interface INavigationElement {
+  children?: INavigationElement[],
+  parent?: string,
+  path?: string
+}
+
+export const NavigationServiceExtension = Symbol("NavigationServiceExtension");
```

Comment:
Just to double check, you're aware that `Symbol("NavigationServiceExtension") == Symbol("NavigationServiceExtension") => false`? AFAIK, If this file is required 'wrong' (uppercase for example), there can be two instances.

Reply:
`packages/singleton.js`:
```
module.exports = new Singleton();
```

`src/part-a.js`:
```
const singleton = require("Singleton");
```

`src/part-b.js`:
```
const singleton = require("singleton");
```

at least in Node 4, with this code you'll end up with two instances of your singleton, because `singleton !== Singleton`

Reply:
We don't care about this class being a singleton anymore. Inversify takes care making it a singleton for us.